File: atcoder/abc143B_takoyaki_festival_2019.ml
(* Vicfred
 * https://atcoder.jp/contests/abc143/tasks/abc143_b
 * brute force
 * *)
Scanf.scanf "%d\n" @@ fun n ->
    let d = Array.init n @@ fun _ -> Scanf.scanf "%d " @@ fun d -> d in
    Printf.printf "%d\n" @@
    Array.fold_left (Array.fold_left ( + )) 0 @@
    Array.init n @@ fun i ->
        Array.init n @@ fun j ->
            if i <= j then 0
            else d.(i) * d.(j)

File: atcoder/abc143B_takoyaki_festival_2019.nim
# unihernandez22
# https://atcoder.jp/contests/abc143/tasks/abc143_b
# greedy

import strutils, sequtils

let
  n = stdin.readLine.parseInt
  d = stdin.readLine.split.map(parseInt)

var
  csum = newSeq[int](n)
  ssum = 0
  ans = 0

for i in 1 .. n:
  csum[n - i] = ssum
  ssum += d[n - i]

for i, elem in d:
  ans += elem * csum[i]

ans.echo
